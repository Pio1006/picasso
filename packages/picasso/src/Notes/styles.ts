// import { PicassoProvider } from '@toptal/picasso-provider'
import { Theme, createStyles } from '@material-ui/core/styles'

// PicassoProvider.override(({ palette }: Theme) => ({}))

export default ({ palette }: Theme) =>
  createStyles({
    root: {
      borderRadius: '0.4rem',
      border: `1px solid ${palette.grey.lighter}`,
      position: 'relative',
      overflow: 'hidden',
      '&:before': {
        background: palette.yellow.main,
        display: 'block',
        height: '100%',
        left: 0,
        position: 'absolute',
        content: '""',
        top: 0,
        width: '0.2rem'
      }
    },
    inner: {
      padding: '1.75rem'
    }
  })
