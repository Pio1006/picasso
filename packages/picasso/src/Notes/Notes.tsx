import React, { forwardRef } from 'react'
import { makeStyles, Theme } from '@material-ui/core/styles'
import { BaseProps, CompoundedComponentWithRef } from '@toptal/picasso-shared'
import cx from 'classnames'

import NotesTitle from '../NotesTitle'
import NotesSubtitle from '../NotesSubtitle'
import NotesContent from '../NotesContent'
import styles from './styles'
import Container from '../Container'

export interface Props extends BaseProps {
  children: React.ReactNode
}

const useStyles = makeStyles<Theme>(styles)

export interface StaticProps {
  Title: typeof NotesTitle
  Subtitle: typeof NotesSubtitle
  Content: typeof NotesContent
}

export const Notes = forwardRef<HTMLDivElement, Props>(function Notes (
  { children, className, ...rest },
  ref
) {
  const classes = useStyles()

  return (
    <Container
      ref={ref}
      bottom='medium'
      className={cx(classes.root, className)}
      {...rest}
    >
      <Container className={classes.inner} flex justifyContent='space-between'>
        <Container>{children}</Container>
      </Container>
    </Container>
  )
}) as CompoundedComponentWithRef<Props, HTMLDivElement, StaticProps>

Notes.defaultProps = {}

Notes.displayName = 'Notes'
Notes.Title = NotesTitle
Notes.Subtitle = NotesSubtitle
Notes.Content = NotesContent

export default Notes
