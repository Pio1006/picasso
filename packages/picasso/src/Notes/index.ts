import { OmitInternalProps } from '@toptal/picasso-shared'

import { Props } from './Notes'

export { default } from './Notes'
export type NotesProps = OmitInternalProps<Props>
