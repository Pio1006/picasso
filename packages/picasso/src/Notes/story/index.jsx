import PicassoBook from '~/.storybook/components/PicassoBook'
import { Notes } from '../Notes'

const page = PicassoBook.section('Components').createPage(
  'Notes',
  `<-- description -->`
)

page
  .createTabChapter('Props')
  .addComponentDocs({ component: Notes, name: 'Notes' })
  .addExample('Notes/story/Default.example.tsx', 'Default')
page.createChapter()
