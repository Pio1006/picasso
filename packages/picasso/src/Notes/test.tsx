import React from 'react'
import { render, RenderResult } from '@toptal/picasso/test-utils'

import Notes from './Notes'

const renderNotes = () => {
  return render(
    <Notes>
      <Notes.Title>Title</Notes.Title>
      <Notes.Subtitle>Subtitle</Notes.Subtitle>
      <Notes.Content>Content</Notes.Content>
    </Notes>
  )
}

describe('Notes', () => {
  let api: RenderResult

  beforeEach(() => {
    api = renderNotes()
  })
  it('default render', () => {
    const { container } = api

    expect(container).toMatchSnapshot()
  })
})
