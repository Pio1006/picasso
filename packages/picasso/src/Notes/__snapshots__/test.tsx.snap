// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes default render 1`] = `
<div>
  <div
    class="Picasso-root"
  >
    <div
      class="PicassoContainer-bottommediumMargin makeStyles-root"
    >
      <div
        class="PicassoContainer-spaceBetweenJustifyContent PicassoContainer-flex makeStyles-inner"
      >
        <div
          class=""
        >
          <div
            class="makeStyles-root"
          >
            <h4
              class="MuiTypography-root PicassoTypography-headingSmall MuiTypography-h4"
            >
              Title
            </h4>
          </div>
          <div
            class="PicassoContainer-bottomsmallMargin PicassoContainer-flex"
          >
            <p
              class="MuiTypography-root PicassoTypography-bodySmall PicassoTypography-darkGrey MuiTypography-body1"
            >
              Subtitle
            </p>
          </div>
          <p
            class="MuiTypography-root PicassoTypography-bodyMedium PicassoTypography-black MuiTypography-body1"
          >
            Content
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
